packages/engine/src/avatar/state/AvatarNetworkState.tsx

```diff
@@ -74,7 +76,31 @@ export const AvatarState = defineState({
         state[action.entityUUID].set(none)
       }
     ]
-  ]
+  ],
+
+  selectRandomAvatar() {
+    Engine.instance.api
+      .service(avatarPath)
+      .find({})
+      .then((avatars: Paginated<AvatarType>) => {
+        const randomAvatar = avatars.data[Math.floor(Math.random() * avatars.data.length)]
+        AvatarState.updateUserAvatarId(randomAvatar.id)
+      })
+  },
+
+  updateUserAvatarId(avatarId: string) {
+    Engine.instance.api
+      .service(userPath)
+      .patch(Engine.instance.userID, { avatarId: avatarId })
+      .then(() => {
+        dispatchAction(
```

Comment:
I dont believe we really need both of these mechanisms. Can we just standardize on listenting for avatar id changes via the feather hooks?

Reply:
this action is what drives our event sourcing, thus it must be persisted to our cached actions